pkg/agent/docker_secrets_postrenderer.go

```diff
@@ -52,6 +52,57 @@ func NewDockerSecretsPostRenderer(secrets map[string]string) (*DockerSecretsPost
 	return r, nil
 }
 
+// resourceListTypes defines all the list types for which we need to
+// search for pod templates.
+var resourceListTypes map[string]struct{} = map[string]struct{}{
+	"CronJobList":               struct{}{},
+	"DaemonSetList":             struct{}{},
+	"DeploymentList":            struct{}{},
+	"JobList":                   struct{}{},
+	"PodTemplateList":           struct{}{},
+	"ReplicaSetList":            struct{}{},
+	"ReplicationControllerList": struct{}{},
+	"StatefulSetList":           struct{}{},
```

Comment:
AFAIK, just `"List"` is also a valid resource type

Reply:
As far as I know, this is only used for the top level which the renderer never sees, but I've added it anyway.